### ISSUE TITLE
Loss function has been implemented wrongly

### ISSUE THREAD

**Issue by Nikronic** [Nikronic/CoarseNet#7]
The loss function of CoarseNet is adapted from VGG style transfer loss which means, first we have to convert Ground Truth Image and Estimated Image to a low spatial tensors, then calculate loss using Gram Matrix and L2 (MSE) error.

We have similar situations in DetailsNet.

Please revise these line in DetailsNet implementation:
[loss.py:58](https://github.com/Nikronic/DetailsNet/blob/af8d51be1575ac180b7e024b6140cf424d5d3697/utils/loss.py#L58)
[vgg.py](https://github.com/Nikronic/DetailsNet/blob/master/vgg.py)

![image](https://user-images.githubusercontent.com/32401852/58167182-9ea90780-7ca0-11e9-9ff7-903e9351b2a4.png)

**Comment by Nikronic** [comment]
Which hidden layer should be chosen as a low-spatial latent vector to pass to loss function?
In the DetailsNet, layers 2 and 5 have been chosen on VGG19BN. But here we are dealing with VGG16BN which has fewer hidden layers.

Note: the last layer of VGGs is classifier so we cannot use them because of 1-D vector representation of the last layer to match with a class of object.

**Comment by Nikronic** [comment]
We merge maxpool_2 and maxpool_ 5 in loss function.
https://github.com/geonm/EnhanceNet-Tensorflow/blob/master/train_SR.py#L90
Note: Before combining them, first we should patch them then calculate loss on merged ones.

**Comment by Nikronic** [comment]
Note we should not apply patch-wise loss. Patch-wise loss only has been used in DetailsNet, but loss function of CoarseNet has been motivated by neural style transfer.![image]
![image](https://user-images.githubusercontent.com/32401852/58551964-81d37d80-8226-11e9-85e2-57fbb458ccc8.png)

Note: in the reference paper of Neural Style Transfer, they used different layers from EnhanceNet.
Note: EnhanceNet patch-loss which uses VGG is required for DetailsNet
Note: NeuralStyleTransfer loss (not patched) which uses VGG is required for CoarseNet
